### PR TITLE
[SDP-877] Use dependency injection to inject the Signature Service into the `serve` and `tss` applications

### PR DESCRIPTION
**Pull request by marcelosalloum**
### What

Use dependency injection to inject the Signature Service into the `serve` and `tss` applications

### Why

As part of https://stellarorg.atlassian.net/browse/SDP-877

#### PR Structure

* [x] This PR has a reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR title and description are clear enough for anyone to review it.
* [x] This PR does not mix refactoring changes with feature changes (split into two PRs otherwise).

#### Thoroughness

* [x] This PR adds tests for the new functionality or fixes.
* [x] This PR contains the link to the Jira ticket it addresses.

#### Configs and Secrets

* [x] No new CONFIG variables are required -OR- the new required ones were added to the helmchart's [`values.yaml`] file.
* [x] No new CONFIG variables are required -OR- the new required ones were added to the deployments ([`pr-preview`], [`dev`], [`demo`], `prd`).
* [x] No new SECRETS variables are required -OR- the new required ones were mentioned in the helmchart's [`values.yaml`] file.
* [x] No new SECRETS variables are required -OR- the new required ones were added to the deployments ([`pr-preview secrets`], [`dev secrets`], [`demo secrets`], `prd secrets`).

#### Release

* [x] This is not a breaking change.
* [ ] **This is ready for production.**. If your PR is not ready for production, please consider opening additional complementary PRs using this one as the base. Only merge this into `develop` or `main` after it's ready for production!

#### Deployment

* [ ] Does the deployment work after merging?

[`values.yaml`]: ../helmchart/sdp/values.yaml
[`pr-preview`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/common-previews/stellar-disbursement-platform/backend-helm-values
[`dev`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/backend-helm-values
[`demo`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/demo/demo-backend-helm-values
[`pr-preview secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/common-previews/externalsecrets-common-previews.yaml#L241-L346
[`dev secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/stellar-disbursement-platform-externalsecrets.yaml
[`demo secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/demo/demo-sdp-externalsecrets.yaml

### CODE REVIEW

File: cmd/serve.go [hunk 1]
```diff
@@ -419,35 +429,49 @@ func (c *ServeCommand) Command(serverService ServerServiceInterface, monitorServ
...
 			if err != nil {
 				log.Ctx(ctx).Fatalf("error creating Anchor Platform API Service: %v", err)
 			}
 			serveOpts.AnchorPlatformAPIService = apAPIService
 
+			// Setup the TSSDBConnectionPool
+			tssDBConnectionPool, err := di.NewTSSDBConnectionPool(ctx, di.TSSDBConnectionPoolOptions{DatabaseURL: globalOptions.DatabaseURL})
+			if err != nil {
+				log.Ctx(ctx).Fatalf("error getting TSS DB connection pool: %v", err)
+			}
```
Comment (CaioTeixeira95) [inline cmd/serve.go:464]: Can we call `defer tssDBConnectionPool.Close()` here?

File: cmd/serve_test.go [hunk 2]
```diff
@@ -125,35 +133,36 @@ func Test_serve(t *testing.T) {
...
 	metricOptions := monitor.MetricOptions{
 		MetricType:  monitor.MetricTypePrometheus,
 		Environment: "test",
 	}
 	mMonitorService.On("Start", metricOptions).Return(nil).Once()
+	defer mMonitorService.AssertExpectations(t)
+
+	encryptionPassphrase := keypair.MustRandom().Seed()
+	serveOpts.SignatureService = engineMocks.NewMockSignatureService(t)
+	di.SetInstance("signature_service_instance", serveOpts.SignatureService)
```
Comment (CaioTeixeira95) [inline cmd/serve_test.go:165]: I loved how you solved the problem of different memory addresses when asserting the `serveOpts` expectations.

File: internal/transactionsubmission/manager.go [hunk 3]
```diff
@@ -79,9 +67,9 @@ func (so *SubmitterOptions) validate() error {
 		return fmt.Errorf("monitor service cannot be nil")
 	}
 
-	if so.EventProducer == nil {
-		return fmt.Errorf("event producer cannot be nil")
-	}
+	// if so.EventProducer == nil {
+	// 	return fmt.Errorf("event producer cannot be nil")
+	// }
```
Comment (CaioTeixeira95) [inline internal/transactionsubmission/manager.go:72]: This was removed on my last merged PR.

File: internal/transactionsubmission/manager_test.go [hunk 4]
```diff
@@ -132,77 +115,54 @@ func Test_SubmitterOptions_validate(t *testing.T) {
...
+		// 		DBConnectionPool:     dbConnectionPool,
+		// 		SignatureService:     mSigService,
+		// 		HorizonURL:           "https://horizon-testnet.stellar.org",
+		// 		NumChannelAccounts:   1,
+		// 		QueuePollingInterval: 10,
+		// 		MaxBaseFee:           txnbuild.MinBaseFee,
+		// 		MonitorService:       tssMonitorService,
+		// 	},
+		// 	wantErrContains: "event producer cannot be nil",
+		// },
```
Comment (CaioTeixeira95) [inline internal/transactionsubmission/manager_test.go:139]: Same here.

File: internal/transactionsubmission/manager.go (hunk 3 above)
Comment (marcelosalloum) [reply internal/transactionsubmission/manager.go:72]: I don't think so. It's still in the sdp-multitenant branch:
https://github.com/stellar/stellar-disbursement-platform-backend/blob/b5a3319740eee01472aa5e99bd5424c6ab89d77c/internal/transactionsubmission/manager.go#L82-L84

File: internal/transactionsubmission/manager_test.go (hunk 4 above)
Comment (marcelosalloum) [reply internal/transactionsubmission/manager_test.go:139]: I don't think so. It's still on the sdp-multitenant branch:
https://github.com/stellar/stellar-disbursement-platform-backend/blob/b5a3319740eee01472aa5e99bd5424c6ab89d77c/internal/transactionsubmission/manager_test.go#L186-L206

File: cmd/serve.go (hunk 1 above)
Comment (marcelosalloum) [reply cmd/serve.go:464]: Done in https://github.com/stellar/stellar-disbursement-platform-backend/pull/158/commits/7c6d5c09156f6f59c9fc68bbf5d7841b286e5dbd.

**Pull request merged**